Urban-Fun/View Controllers/SearchViewController.m

```diff
@@ -6,26 +6,225 @@
+        searchText = [searchText stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]];
+        PFQuery *userQuery = [User query];
+        [userQuery whereKey:@"username" matchesRegex:searchText modifiers:@"i"];
+        [userQuery includeKey:@"host"];
+        self.usersArray = [[NSMutableArray alloc] init];
+        [userQuery findObjectsInBackgroundWithBlock:^(NSArray<User *>*users , NSError *error) {
+            if (error) {
+                
+            }
+            for (User *user in users) {
+                if (![self.usersArray containsObject:user]){
+                    [self.usersArray addObject:user];
+                }
+            }
+            [self.tableView reloadData];
+
+        }];
+    }
+}
+
+- (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText {
+    if (self.searchControl.selectedSegmentIndex == 0){
+        //removing white space
+        searchText = [searchText stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]];
+        PFQuery *activityQuery = [Activity query];
+        [activityQuery whereKey:@"title" matchesRegex:searchText modifiers:@"i"];
+        [activityQuery orderByDescending:@"createdAt"];
+        [activityQuery includeKey:@"host"];
+        self.activitiesArray = [[NSMutableArray alloc] init];
+        [activityQuery findObjectsInBackgroundWithBlock:^(NSArray<Activity *>*activities , NSError *error) {
+            if (error) {
+                
+            }
+            for (Activity *activity in activities) {
+                if (![self.activitiesArray containsObject:activity]){
+                    [self.activitiesArray addObject:activity];
+                }
+            }
+            [self.tableView reloadData];
+        }];
+    }
+    if (self.searchControl.selectedSegmentIndex == 1){
+        //removing white space
+        searchText = [searchText stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]];
+        PFQuery *userQuery = [User query];
+        [userQuery whereKey:@"username" matchesRegex:searchText modifiers:@"i"];
+        [userQuery includeKey:@"host"];
+        self.usersArray = [[NSMutableArray alloc] init];
+        [userQuery findObjectsInBackgroundWithBlock:^(NSArray<User *>*users , NSError *error) {
+            if (error) {
+                
+            }
+            for (User *user in users) {
+                if (![self.usersArray containsObject:user]){
+                    [self.usersArray addObject:user];
+                }
+            }
+            [self.tableView reloadData];
+        }];
+    }
+    
+}
+
+// delegate method to pass tapped user from cell
+- (void)didTapUsername:(nonnull User *)user {
+    self.profileToView = user;
+}
+// action method
+- (IBAction)usernameTapped:(id)sender {
+    [self performSegueWithIdentifier:@"profileFromSearch" sender:sender];
+}
+
+- (void)syncButtons {
+    [self.tableView reloadData];
```

Comment:
Is there a different way to achieve this? You can add this to your polish week, there should be a way to avoid reloading all the table view if you only changed one item